components/panelbar/events.md

```diff
@@ -0,0 +1,307 @@
+
+The event handler receives a `PanelBarItemClickEventArgs` object which provides the model of the clicked node in the `Item` field that you can cast to your model type.
+
+If that item has children the `OnItemClick` event will fire before the `ExpandedItemsChanged` event.
+
+>caption Handle OnItemClick to react to user click action.
+
+````CSHTML
+@* Use the OnItemClick event to react to the user click action. *@
+
+<div style="width: 30%;">
+    <TelerikPanelBar Data="@Items" OnItemClick="@OnItemClickHandler">
+    </TelerikPanelBar>
+</div>
+
+@code {
+    public List<PanelBarItem> Items { get; set; }
+
+    private void OnItemClickHandler(PanelBarItemClickEventArgs args)
+    {
+        Console.WriteLine($"The clicked item is: {(args.Item as PanelBarItem).Text}");
+    }
+
+    public class PanelBarItem
+    {
+        public int Id { get; set; }
+        public string Text { get; set; }
+        public int? ParentId { get; set; }
+        public bool HasChildren { get; set; }
+    }
+
+    private List<PanelBarItem> LoadFlatData()
+    {
+        List<PanelBarItem> items = new List<PanelBarItem>();
+
+        items.Add(new PanelBarItem()
+        {
+            Id = 1,
+            Text = "Company",
+            ParentId = null,
+            HasChildren = false,
+        });
+
+        items.Add(new PanelBarItem()
+        {
+            Id = 2,
+            Text = "Contact us",
+            ParentId = null,
+            HasChildren = true,
+        });
+
+        items.Add(new PanelBarItem()
+        {
+            Id = 3,
+            Text = "Email",
+            ParentId = 2,
+            HasChildren = false
+        });
+
+        items.Add(new PanelBarItem()
+        {
+            Id = 4,
+            Text = "LinkedIn",
+            ParentId = 2,
+            HasChildren = false
+        });
+
+        items.Add(new PanelBarItem()
+        {
+            Id = 5,
+            Text = "Audio",
+            ParentId = null,
+            HasChildren = false
+        });
+
+        return items;
+    }
+
+    protected override void OnInitialized()
+    {
+        Items = LoadFlatData();
+
+        base.OnInitialized();
+    }
+}
+````
+
+## ExpandedItemsChanged
+
+The `ExpandedItemsChanged` event fires when the user expands or collapses a node (item) in the PanelBar.
+
+>caption Handle the ExpandedItemsChanged events
+
+````CSHTML
+@* Use the ExpandedItemsChanged to respond to the user expanding or collapsing items in the PanelBar *@
+
+<div style="width: 30%;">
+    <TelerikPanelBar Data="@Items" ExpandedItems="@ExpandedItems" ExpandedItemsChanged="@ExpandedItemsChanged">
+    </TelerikPanelBar>
+</div>
+
+@code {
+    public List<PanelBarItem> Items { get; set; }
+
+    public IEnumerable<object> ExpandedItems { get; set; } = new List<object>();
+
+    private void ExpandedItemsChanged(IEnumerable<object> expandedItems)
```

Comment:
May be it is worth to add something to show the event firing - a console.writeline or rendering a list of the expanded items?

Reply:
I am not sure what value this CW would bring, because this event comes from the framework and people should be aware of the firing timing, although I have added a line to showcase that.